hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go

```diff
@@ -63,7 +63,24 @@ func (p AWS) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, createOr
 }
 
 func (p AWS) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
-	providerImage := imageCAPA
+	var pullSecret corev1.Secret
+	ctx := context.TODO()
+	c, err := cmdUtil.GetClient()
```

Comment:
We have a pattern already to pass images
see
```
type AWS struct {
	utilitiesImage string
}
```

Let's add a field for the caprovider images or consider passing the image as a parameter in the interface signature

Reply:
Not sure what you mean here, but did my recent changes make this comment OBE? 

If not, would you please explain a little further so I can understand what's being asked?

Reply:
rather can enable a function that create a client everytime, let's reuse our shared existing instantiated once only clients. We what we do with utilitiesImage in New, you could do the same with the caprovider images and let getPlatform to find the image for each platform

Reply:
I'm sorry. I understand more but I don't think all the way. Are you suggesting to look up the capiProviderImage within `hostedcluster_controller.go` and then pass it to `platform.GetPlatform(...)`?
<br/>

> let getPlatform to find the image for each platform

This is the part I'm confused on. I thought you were suggesting to look up the capiProviderImage within `platform.go` and reuse `GetPayloadImage` and `getPullSecretFromHostedCluster` from the updated `imagemetadata.go` in this PR but I also thought you were asking me to avoid using `cmdUtil.GetClient()`, which I do in `getPullSecretFromHostedCluster`.

Reply:
Pleased don't use cmdUtil.GetClient() within a func. 

>@sjenning I think this might be due to the change I made in order to delete the AWS machine sets. I'll try reworking that and see if that resolves the problem.

The recreated every loop non cached client is most likely causing that.

>Are you suggesting to look up the capiProviderImage within hostedcluster_controller.go and then pass it to platform.GetPlatform(...)

Either that or doing it from within `GetPlatform` func itself, there's many choices.

Reply:
Ok will do